eng/tools/doccheck/main.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/doc"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func filter(f fs.FileInfo) bool {
+	return !strings.HasSuffix(f.Name(), "_test.go")
+}
+
+func findAllSubDirectories(root string) []string {
+	ret := make([]string, 0)
```

Comment:
Small nit, but you can just do this:

```go
var ret []string
```

A 'nil' slice value is basically an empty slice - append (and len) handle it just fine.